contracts/adapters/TributeNFT.sol

```diff
@@ -0,0 +1,267 @@
+pragma solidity ^0.8.0;
+
+// SPDX-License-Identifier: MIT
+
+import "./interfaces/ITribute.sol";
+import "../core/DaoConstants.sol";
+import "../core/DaoRegistry.sol";
+import "../extensions/nft/NFT.sol";
+import "../adapters/interfaces/IVoting.sol";
+import "../guards/MemberGuard.sol";
+import "../guards/AdapterGuard.sol";
+import "../utils/IERC20.sol";
+import "../helpers/AddressLib.sol";
+import "../helpers/SafeERC20.sol";
+
+/**
+MIT License
+
+Copyright (c) 2020 Openlaw
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+ */
+
+contract TributeNFTContract is
+    ITribute,
+    DaoConstants,
+    MemberGuard,
+    AdapterGuard
+{
+    using Address for address;
+
+    struct ProposalDetails {
+        // The proposal id.
+        bytes32 id;
+        // The applicant address (who will receive the DAO Shares and become a member).
+        address applicant;
+        // The address of the NFT to be locked in the DAO in exchange for voting power.
+        address nftAddr;
+        // The nft token identifier.
+        uint256 nftTokenId;
+        // The amount of shares requested to lock the NFT in exchange of DAO internal tokens.
+        uint256 requestedShares;
+    }
+
+    // Keeps track of all nft tribute proposals handled by each DAO.
+    mapping(address => mapping(bytes32 => ProposalDetails)) public proposals;
+
+    /**
+     * @notice default fallback function to prevent from sending ether to the contract.
+     */
+    receive() external payable {
+        revert("fallback revert");
+    }
+
+    function configureDao(DaoRegistry dao, address nftToCollect)
+        external
+        onlyAdapter(dao)
+    {
+        NFTExtension nftExt = NFTExtension(dao.getExtensionAddress(NFT));
+        nftExt.registerPotentialNewNFT(nftToCollect);
+    }
+
+    function provideTribute(
```

Comment:
we do not need this function there right ?

Reply:
We don't need, but as we are using the ITribute interface it requires an implementation, so I added a dummy one with the revert call to prevent any usages.